src/linux/wifi/core/AccessPointControllerLinux.cxx

```diff
@@ -197,7 +285,72 @@ AccessPointControllerLinux::SetProtocol(Microsoft::Net::Wifi::Ieee80211Protocol
     }
 
     // Reload hostapd conf file.
-    return isOk && m_hostapd.Reload();
+    isOk = isOk && m_hostapd.Reload();
+
+    return isOk;
+}
+
+bool
+AccessPointControllerLinux::SetAkmSuites(std::vector<Ieee80211AkmSuite> akmSuites)
+{
+    bool isOk = false;
+
+    // Construct the wpa_key_mgmt hostapd property value.
+    std::string concatenatedWpaKeyMgmtPropertyValue{};
+    for (const auto& akmSuite : akmSuites) {
+        const auto& wpaKeyMgmtPropertyValue = detail::IeeeAkmSuiteToWpaKeyMgmtPropertyValue(akmSuite);
+        concatenatedWpaKeyMgmtPropertyValue += wpaKeyMgmtPropertyValue;
+        concatenatedWpaKeyMgmtPropertyValue += " ";
+    }
+
+    try {
+        // Set the hostapd wpa_key_mgmt property.
+        isOk = m_hostapd.SetProperty(Wpa::ProtocolHostapd::PropertyNameWpaKeyMgmt, concatenatedWpaKeyMgmtPropertyValue);
+
+        // For simplicity, set the hostapd wpa property to allow WPA and WPA2 (and WPA3).
+        isOk = isOk && m_hostapd.SetProperty(Wpa::ProtocolHostapd::PropertyNameWpa, Wpa::ProtocolHostapd::PropertyWpaValueWpaAndWpa2);
+
+        // For simplicity, set the hostapd auth_algs property to allow both Open System Authentication and Shared Key Authentication.
+        isOk = isOk && m_hostapd.SetProperty(Wpa::ProtocolHostapd::PropertyNameAuthAlgs, Wpa::ProtocolHostapd::PropertyAuthAlgsValueOpenAndSharedKey);
+    } catch (const Wpa::HostapdException& ex) {
+        throw AccessPointControllerException(std::format("Failed to set AKM suites for interface {} ({})", GetInterfaceName(), ex.what()));
+    }
+
+    // Reload hostapd conf file.
+    isOk = isOk && m_hostapd.Reload();
+
+    return isOk;
+}
+
+bool
+AccessPointControllerLinux::SetCipherSuites(std::vector<Ieee80211CipherSuite> cipherSuites)
+{
+    bool isOk = false;
+
+    // TODO: Handle WEP separately.
+
+    // For simplicity, construct both wpa_pairwise and rsn_pairwise hostapd property values.
+    std::string concatenatedPairwisePropertyValue{};
+    for (const auto& cipherSuite : cipherSuites) {
+        const auto& pairwisePropertyValue = detail::IeeeCipherSuiteToPairwisePropertyValue(cipherSuite);
+        concatenatedPairwisePropertyValue += pairwisePropertyValue;
+        concatenatedPairwisePropertyValue += " ";
+    }
+
+    try {
+        // Set the hostapd wpa_pairwise property.
+        isOk = m_hostapd.SetProperty(Wpa::ProtocolHostapd::PropertyNameWpaPairwise, concatenatedPairwisePropertyValue);
+
+        // Set the hostapd rsn_pairwise property.
+        isOk = isOk && m_hostapd.SetProperty(Wpa::ProtocolHostapd::PropertynameRsnPairwise, concatenatedPairwisePropertyValue);
```

Comment:
Just want to confirm that both `WpaPairWise` and `RsnPairWise` are supposed to get the same value here? Also, it looks like `PropertynameRsnPairwise` should be renamed to `PropertyNameRsnPairwise` (lowercase 'n' -> 'N').

Reply:
The default value for `rsn_pairwise` in hostapd is the value of `wpa_pairwise`, so I just set them both to the same value for simplicity. However, they could technically be set with different values, so perhaps this should be changed. And good catch!